### PR TITLE
Using github action to update the version number according to the tag

### PR DESCRIPTION
**Pull request by PaypayBob**
This PR has for goal to use github action to update the version number according to the tag when a release is created.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing](/contributing.md) document?
* [x] Have you read and signed the automated Contributor's License Agreement?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### New Feature Submissions:

1. [NA] Does your submission pass tests?
2. [] Have you lint your code locally prior to submission?

### CODE REVIEW

File: .github/workflows/python-publish.yml [hunk 1]
```diff
@@ -14,6 +14,24 @@ jobs:
...
+    - name: Commit the changes
+      run: |
+      git add ./setup.py
+      git config --local user.email "action@github.com"
+      git config --local user.name "GitHub Action"
+      git commit -m "ci: Automatic version bump" -a | exit 0
+    - name: Push the changes
+      uses: ad-m/github-push-action@master
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (shreyanshp) [inline .github/workflows/python-publish.yml:34]: Why does it have to commit and push back?

File: .github/workflows/python-publish.yml [hunk 2]
```diff
@@ -14,6 +14,24 @@ jobs:
...
     steps:
     - uses: actions/checkout@v2
+      with:
+        persist-credentials: false
+        fetch-depth: 0
+    - name: Set version veriable
+      id: vars
+      run: echo ::set-env name=RELEASE_VERSION::${GITHUB_REF#refs/*/}
+    - name: Update versiona rel
+      run: sed -i -E "s/version=\\\"[0-9.]*\\\",/version=\\\"$RELEASE_VERSION\\\",/g" ./setup.py
```
Comment (shreyanshp) [inline .github/workflows/python-publish.yml:24]: You can actually update the setup.py to have a `placeholder` for version and replace that here and then just build and deploy

**Review by shreyanshp — CHANGES_REQUESTED** [review]

**Comment by PaypayBob** [comment]
Ok to answer to both question at the same time, I thought that we wanted the repo to have the update version of the setup.py file, if we don't need, then yes I can create a placeholder and not commit/push the modification ! It should be totally fine for the build and release

File: .github/workflows/python-publish.yml (hunk 1 above)
Comment (PaypayBob) [reply .github/workflows/python-publish.yml:34]: Answered here : https://github.com/paypay/paypayopa-sdk-python/pull/29#issuecomment-705312974

File: .github/workflows/python-publish.yml (hunk 2 above)
Comment (PaypayBob) [reply .github/workflows/python-publish.yml:24]: Answered here : https://github.com/paypay/paypayopa-sdk-python/pull/29#issuecomment-705312974

**Review by solancer — APPROVED** [review]
LGTM

**Review by shreyanshp — APPROVED** [review]

**Comment by shreyanshp** [comment]
Thank you @PaypayBob !

**Pull request merged**